### PR TITLE
[MaterialButton] Align icon on top

### LINKED ISSUE #671: [MaterialButton] Add support for TOP & BOTTOM icon gravity
**Is your feature request related to a problem? Please describe.**
Up until release `1.1.0-alpha06` back in May 2019, the only way to change the icon position was done using  `TextViewCompat.setCompoundDrawablesRelative`.

After introducing new method `setIconGravity` in release `1.1.0-alpha06`  it was finally possible to properly set the icon position, although only on the right or left, and close to the text or not.

**This now excludes the possibility to place the icon either on top or bottom relative to the text.**

**Describe the solution you'd like**
Please kindly add TOP & BOTTOM gravity as part of `@IconGravity`. 

**Describe alternatives you've considered**
As said at the beginning, before release `1.1.0-alpha06`, the only way to change the icon position was using `TextViewCompat.setCompoundDrawablesRelative`. 

This no longer works, as it triggers a `requestLayout` which, at some point, will trigger `updateIconPosition` and then `updateIcon` where only START/TEXT & END/TEXT are supported.

### PR DESCRIPTION
**Pull request by rewgoes**
closes #671 

According to the discussion in #671, we got approval from design to support adding gravity Top.

### CODE REVIEW

File: lib/java/com/google/android/material/button/MaterialButton.java [hunk 1]
```diff
@@ -666,35 +682,49 @@ private void updateIcon(boolean needsIconUpdate) {
...
     if (needsIconUpdate) {
-      resetIconDrawable(isIconStart);
+      resetIconDrawable();
       return;
     }
 
     // Otherwise only update if the icon or the position has changed
     Drawable[] existingDrawables  = TextViewCompat.getCompoundDrawablesRelative(this);
     Drawable drawableStart = existingDrawables[0];
+    Drawable drawableTop = existingDrawables[0];
```
Comment (antonkovalyov351) [inline lib/java/com/google/android/material/button/MaterialButton.java]: existingDrawables[1]

File: lib/java/com/google/android/material/button/MaterialButton.java [hunk 2]
```diff
@@ -157,8 +157,22 @@
    */
   public static final int ICON_GRAVITY_TEXT_END = 0x4;
 
+  /**
+   * Gravity used to position the icon in the center of the view on top of the text
+   *
+   * @see #setIconGravity(int)
+   * @see #getIconGravity()
+   */
+  public static final int ICON_GRAVITY_TOP = 0x5;
```
Comment (antonkovalyov351) [inline lib/java/com/google/android/material/button/MaterialButton.java]: Can you please also add support for ICON_GRAVITY_TEXT_TOP?

File: lib/java/com/google/android/material/button/MaterialButton.java [hunk 3]
```diff
@@ -157,8 +157,22 @@
    */
   public static final int ICON_GRAVITY_TEXT_END = 0x4;
 
+  /**
+   * Gravity used to position the icon in the center of the view on top of the text
```
Comment (antonkovalyov351) [inline lib/java/com/google/android/material/button/MaterialButton.java]: It is not really on top of the text.
Should be: `Gravity used to position the icon at the top of the view.`

File: lib/java/com/google/android/material/button/MaterialButton.java (hunk 2 above)
Comment (rewgoes) [reply lib/java/com/google/android/material/button/MaterialButton.java]: Can you please explain the difference between ICON_GRAVITY_TOP  and ICON_GRAVITY_TEXT_TOP?

If the idea is to keep the icon closer to the top of the view (ICON_GRAVITY_TOP) or closer to the top of the text (ICON_GRAVITY_TEXT_TOP), in my point of view, both can be achieved by adjusting iconPadding and paddingTop.

File: lib/java/com/google/android/material/button/MaterialButton.java (hunk 3 above)
Comment (rewgoes) [reply lib/java/com/google/android/material/button/MaterialButton.java]: Fixed by be01462

File: lib/java/com/google/android/material/button/MaterialButton.java (hunk 1 above)
Comment (rewgoes) [reply lib/java/com/google/android/material/button/MaterialButton.java]: Fixed by f376603

File: lib/java/com/google/android/material/button/MaterialButton.java (hunk 2 above)
Comment (antonkovalyov351) [reply lib/java/com/google/android/material/button/MaterialButton.java]: IMO adjusting `iconPadding` along with `paddingTop` is hacky and not flexible enough.

Comment (rewgoes) [reply lib/java/com/google/android/material/button/MaterialButton.java]: Thanks for the suggestion, after adding the support to textTop and playing around with the feature I've noticed that it's really handy to have it. 👍 

Solution add on d43aa24

**Comment by rewgoes** [comment]
Thanks for approving that. Should I do something next or just wait?

**Comment by rewgoes** [comment]
@ymarian do you know if someone can review this PR?

**Comment by hunterstich** [comment]
@rewgoes This looks great, thanks! Nothing else is needed at the moment. I'm working on getting this merged and it should be synced out soon

**Comment by rewgoes** [comment]
@hunterstich , is there any update on this new feature?

**Comment by rewgoes** [comment]
@raajkumars is there any plan to merge this PR?

**Comment by hunterstich** [comment]
Hey @rewgoes, sorry for the silence. I've been working on making some small changes, adding tests and getting this submitted. I'm running into a couple issues merging internally but I'm working on it and it should be synced out next week.

**Comment by leticiarossi** [comment]
There's been some internal issues that weren't caught before so this has to be reopened

**Comment by rewgoes** [comment]
Is there anything that I can help? What was this issue?

**Comment by rewgoes** [comment]
@leticiarossi and @dsn5ft , what is the internal issue that you found?

I believe we've lost the window to release the feature on version 1.2.0, as the beta version was already released.

**Pull request closed without merging**